consensus/src/liveness/proposal_generator.rs

```diff
@@ -135,73 +125,7 @@ impl ProposalGenerator {
-                        WaitingSuccess::WaitWasRequired {
-                            current_duration_since_epoch,
-                            wait_duration,
-                        } => {
-                            counters::PROPOSAL_SUCCESS_WAIT_S.observe_duration(wait_duration);
-                            counters::PROPOSALS_GENERATED_COUNT
-                                .with_label_values(&["wait_was_required"])
-                                .inc();
-                            current_duration_since_epoch
-                        }
-                        WaitingSuccess::NoWaitRequired {
-                            current_duration_since_epoch,
-                            ..
-                        } => {
-                            counters::PROPOSAL_SUCCESS_WAIT_S.observe_duration(Duration::new(0, 0));
-                            counters::PROPOSALS_GENERATED_COUNT
-                                .with_label_values(&["no_wait_required"])
-                                .inc();
-                            current_duration_since_epoch
-                        }
-                    }
-                }
-                Err(waiting_error) => {
-                    match waiting_error {
-                        WaitingError::MaxWaitExceeded => {
-                            counters::PROPOSAL_FAILURE_WAIT_S.observe_duration(Duration::new(0, 0));
-                            counters::PROPOSALS_GENERATED_COUNT
-                                .with_label_values(&["max_wait_exceeded"])
-                                .inc();
-                            bail!(
-                                "Waiting until parent block timestamp usecs {:?} would exceed the round duration {:?}, hence will not create a proposal for this round",
-                                hqc.certified_block().timestamp_usecs(),
-                                round_deadline);
-                        }
-                        WaitingError::WaitFailed {
-                            current_duration_since_epoch,
-                            wait_duration,
-                        } => {
-                            counters::PROPOSAL_FAILURE_WAIT_S.observe_duration(wait_duration);
-                            counters::PROPOSALS_GENERATED_COUNT
-                                .with_label_values(&["wait_failed"])
-                                .inc();
-                            bail!(
-                                "Even after waiting for {:?}, parent block timestamp usecs {:?} >= current timestamp usecs {:?}, will not create a proposal for this round",
-                                wait_duration,
-                                hqc.certified_block().timestamp_usecs(),
-                                current_duration_since_epoch);
-                        }
-                    };
-                }
-            }
-        };
+        let block_timestamp = self.time_service.get_current_timestamp();
```

Comment:
```suggestion
        // All proposed blocks in a branch are guaranteed to have increasing timestamps
        // since their predecessor block will not be added to the BlockStore until 
        // the local time exceeds it.
        let block_timestamp = self.time_service.get_current_timestamp();
```